### PR TITLE
IGNITE-20519 Update token for CatalogObjectDescriptor is introduced

### PR DESCRIPTION
**Pull request by alievmirza**
https://issues.apache.org/jira/browse/IGNITE-20519

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerDescriptorCausalityTokenTest.java [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+    public void testRenameZone() throws InterruptedException {
+        String zoneName = ZONE_NAME + 1;
+
+        CreateZoneParams createParams = CreateZoneParams.builder().zoneName(zoneName).build();
+
+        assertThat(manager.createZone(createParams), willCompleteSuccessfully());
+
+        long beforeDropTimestamp = clock.nowLong();
+
+        Thread.sleep(5);
```
Comment (sergeyuttsel) [inline modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerDescriptorCausalityTokenTest.java:378]: Is it needed?

Comment (alievmirza) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerDescriptorCausalityTokenTest.java]: Removed

File: modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerDescriptorCausalityTokenTest.java [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+
+        assertNotNull(schema);
+        assertEquals(SCHEMA_NAME, schema.name());
+        assertSame(schema, manager.activeSchema(123L));
+        assertEquals(INITIAL_CAUSALITY_TOKEN, schema.updateToken());
+
+        assertNull(schema.table(TABLE_NAME));
+        assertNull(manager.table(TABLE_NAME, 123L));
+
+        // Validate actual catalog
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/catalog/src/test/java/org/apache/ignite/internal/catalog/CatalogManagerDescriptorCausalityTokenTest.java]: fixed

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogObjectDescriptor.java [hunk 3]
```diff
@@ -52,6 +64,30 @@ public Type type() {
         return type;
     }
 
+
+    /**
+     * Token of the update of the descriptor.
+     * Updated when {@link UpdateEntry#applyUpdate(org.apache.ignite.internal.catalog.Catalog, long)} is called for the
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogObjectDescriptor.java:63]: Updated

File: modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogObjectDescriptor.java [hunk 4]
```diff
@@ -30,11 +33,20 @@ public abstract class CatalogObjectDescriptor implements Serializable {
     private final int id;
     private final String name;
     private final Type type;
+    private long lastUpdateToken;
 
     CatalogObjectDescriptor(int id, Type type, String name) {
```
[the remark this answers is not in the document]
Comment (alievmirza) [reply modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogObjectDescriptor.java]: Removed

**Review by ibessonov — APPROVED** [review]

**Pull request merged**